### PR TITLE
HTML: tests for <input type=checkbox switch>

### PR DESCRIPTION
**Pull request by annevk**
For https://github.com/whatwg/html/pull/9546.

### CODE REVIEW

**Review by cookiecrook — COMMENTED** [review]
Please add a computedrole subtest in `/html-aam/roles.html`

File: html-aam/roles.html [hunk 1]
```diff
@@ -93,6 +93,7 @@ <h6 data-testname="el-h6" data-expectedrole="heading" class="ex">x</h6>
 
 <input type="button" value="x" data-testname="el-input-button" data-expectedrole="button" class="ex">
 <input type="checkbox" data-testname="el-input-checkbox" data-expectedrole="checkbox" class="ex">
+<input type="checkbox" switch data-testname="el-input-checkbox-switch" data-expectedrole="switch" class="ex">
```
Comment (cookiecrook) [inline html-aam/roles.html:96]: @scottaohara does that ID align with the hash link you intend to use in the HTML-AAM PR? If not, please suggest a replacement. The test names are intended to align with the spec IDs.

Comment (cookiecrook) [reply html-aam/roles.html:96]: FWIW, `el-input-checkbox-switch` looks good to me.

Comment (scottaohara) [reply html-aam/roles.html:96]: This is what I would have used. So lgtm

File: html/rendering/widgets/input-checkbox-switch-notref.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<!doctype html>
+<input type=checkbox>
+<input type=checkbox>
+<input type=checkbox>
```
Comment (nt1m) [inline html/rendering/widgets/input-checkbox-switch-notref.html:4]:
```suggestion
<input type=checkbox>
<input type=checkbox>
<input type=checkbox switch>
```
The third one can have the switch attribute, since the test file doesn't have it on the third one.

File: html-aam/roles.html [hunk 3]
```diff
@@ -203,4 +203,4 @@ <h6 data-testname="el-h6" data-expectedrole="heading" class="ex">x</h6>
 </script>
 
 </body>
-</html>
\ No newline at end of file
+</html></html>
```
Comment (nt1m) [inline html-aam/roles.html:206]: Seems unrelated?

File: html/semantics/selectors/pseudo-classes/indeterminate-switch.indeterminate.window.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const input = document.body.appendChild(document.createElement("input"));
+  t.add_cleanup(() => input.remove());
+  input.type = "checkbox";
+  input.indeterminate = true;
+
+  assert_true(input.matches(":indeterminate"));
+
+  input.setAttribute("switch", "blah");
+  assert_false(input.matches(":indeterminate"));
+}, "Checkbox that becomes a switch control does not match :indeterminate");
```
Comment (nt1m) [inline html/semantics/selectors/pseudo-classes/indeterminate-switch.indeterminate.window.js:34]: Similar invalidation tests for `:has(:indeterminate)` would be nice (:has() invalidation requires some code)

See LayoutTests/imported/w3c/web-platform-tests/css/selectors/invalidation/input-pseudo-classes-in-has.html

File: html/semantics/selectors/pseudo-classes/input-checkbox-switch.tentative.window.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  const input = document.body.appendChild(document.createElement("input"));
+  t.add_cleanup(() => input.remove());
+  input.type = "checkbox";
+  input.indeterminate = true;
+
+  assert_true(body.matches(":has(:indeterminate)"));
+
+  input.switch = true;
+  assert_false(body.matches(":has(:indeterminate)"));
+}, "Parent of a checkbox that becomes a switch control does not match :has(:indeterminate)");
```
Comment (nt1m) [inline html/semantics/selectors/pseudo-classes/input-checkbox-switch.tentative.window.js:46]: Testing style invalidation (not just matching) in a separate commit like css/selectors/invalidation/input-pseudo-classes-in-has.html would be valuable. Having the styling stay stale is a common issue with :has()

Comment (nt1m) [reply html/semantics/selectors/pseudo-classes/input-checkbox-switch.tentative.window.js]: css/selectors/invalidation/media-pseudo-classes-in-has.html is another good example to go off

Comment (annevk) [reply html/semantics/selectors/pseudo-classes/input-checkbox-switch.tentative.window.js]: That would have been nice to know an hour ago. I think this can be a follow-up.

**Review by nt1m — APPROVED** [review]
I haven't reviewed the AX tests, but the HTML ones look fine
